source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -8,13 +8,17 @@ namespace Filters {
 namespace Common {
 namespace RBAC {
 
-MatcherConstSharedPtr Matcher::create(const envoy::config::rbac::v2alpha::Permission& permission) {
+MatcherConstSharedPtr Matcher::create(const envoy::config::rbac::v2alpha::Permission& permission,
+                                      bool disable_http_rules) {
   switch (permission.rule_case()) {
   case envoy::config::rbac::v2alpha::Permission::RuleCase::kAndRules:
-    return std::make_shared<const AndMatcher>(permission.and_rules());
+    return std::make_shared<const AndMatcher>(permission.and_rules(), disable_http_rules);
   case envoy::config::rbac::v2alpha::Permission::RuleCase::kOrRules:
-    return std::make_shared<const OrMatcher>(permission.or_rules());
+    return std::make_shared<const OrMatcher>(permission.or_rules(), disable_http_rules);
   case envoy::config::rbac::v2alpha::Permission::RuleCase::kHeader:
+    if (disable_http_rules) {
```

Comment:
Thinking about this again, why do we need `disable_http_rules` and ignore HTTP headers? When a HeaderMatch specified in TCP filter, shouldn't it just reject the configuration?

Reply:
It makes sense for me to treat this simply as a deny, @liminw WDYT?

Reply:
Agree. If a rules says "allow requests with header a=foo", the rules does not match in TCP filter.

Reply:
Updated to reject invalid rules in `config.cc`